### DISCUSSION TITLE
Consiglio approccio per ciclo scene in un'area su singolo pulsante

### DISCUSSION THREAD

**Discussion by fabri-it** [indomus/forum#2360] [HUB: Home Assistant]
Salve a tutti, vorrei aver conferma di utilizzare un approccio corretto:

ho diversi punti luce in una singola area e relative scene già impostate per definire i diversi stati di ciascuno di essi a seconda delle esigenze. Vorrei, utilizzando un singolo switch, passare ciclicamente fra una scena e l'altra.
Ho pensato di 
- definire un contatore di stato (default a 0, incremento di 1) che definisca la scena attiva (0 = tutto spento; 1...X = una delle scene attive).
- definire un'automazione che al long_press dello switch resetti il contatore di stato a 0 e spenga tutto nell'area
- definire una seconda automazione che, condizionata dal valore del contatore di stato, attivi al single press dello switch una specifica scena e contestualmente aggiunga 1 al contatore fino al raggiungimento del valore massimo e, al successivo single press, spenga tutta l'area e lo resetti al valore di default (0).

Chiedo: è l'approccio corretto? Esistono alternative più eleganti?
Grazie a quanti riterranno opportuno darmi i loro consigli/pareri.

**Comment by 50494554524F** [comment]
come contatore puoi usare un "input_select" così da poterlo utilizzare anche nell'interfaccia nel caso volessi selezionare una scena o usare pulsanti, per il resto farei un'automazione che si occupa di gestire il contatore e un'altra di cambiare la scena (divisione logica dei compiti)

**Reply by fabri-it** [reply]
Grazie del consiglio.
